src/layers/models/layersManager.ts

```diff
@@ -106,28 +144,49 @@ class LayersManager {
     this.logger.info(`Successfully updated mosaic: '${mosaicName}'`);
   }
 
+  public getAllLinkedCaches(baseCacheNames: string[], mapproxyConfiguration: IMapProxyJsonDocument): string[] {
+    const linkedCaches: string[] = [];
+    const baseCacheNamesDuplicate: string[] = [...baseCacheNames];
+    baseCacheNamesDuplicate.forEach((currentCache) => {
+      if (mapproxyConfiguration.caches[currentCache]) {
+        if (currentCache.endsWith('-source')) {
+          if (linkedCaches.indexOf(currentCache) == -1) {
+            linkedCaches.push(currentCache);
+          }
+        } else {
+          linkedCaches.push(currentCache);
+          if (linkedCaches.indexOf(`${currentCache}-source`) == -1) {
+            linkedCaches.push(`${currentCache}-source`);
+          }
+        }
+      }
+    });
+    return linkedCaches;
+  }
+
   public async removeLayer(layersName: string[]): Promise<string[] | void> {
     this.logger.info(`Remove layers request for: [${layersName.join(',')}]`);
-    const failedLayers: string[] = [];
     const errorMessage = 'no valid layers to delete';
+    let failedLayers;
 
     const editJson = (jsonDocument: IMapProxyJsonDocument): IMapProxyJsonDocument => {
       let updateCounter = 0;
-      layersName.forEach((layerName) => {
+      const allLinkedCaches = this.getAllLinkedCaches(layersName, jsonDocument);
+      allLinkedCaches.forEach((cacheName) => {
         // remove requested layer cache source from cache list
-        delete jsonDocument.caches[layerName];
+        delete jsonDocument.caches[cacheName];
         // remove requested layer from layers array
-        const requestedLayerIndex: number = jsonDocument.layers.findIndex((layer) => layer.name === layerName);
-        const negativeResult = -1;
-        if (requestedLayerIndex !== negativeResult) {
+        const requestedLayerIndex: number = jsonDocument.layers.findIndex((layer) => layer.name === cacheName);
+        if (requestedLayerIndex !== -1) {
```

Comment:
Why don't you use other methods such as incldes and find? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find